File: PREPARCIAL/Seqüències/suma-digits.cc
#include <iostream>
using namespace std;
int main()
{
    int a, b;
    while (cin >> a) {
        int x = a;
        int i = 0;
        while (x > 0) {
            b = x % 10;
            if (b != 0) {
                i = i + b;
            }
            x = x / 10;
        }
        cout << "La suma dels digits de " << a << " es " << i << "." << endl;
    }
}

File: PREPARCIAL/Python/suma-digits.py
from easyinput import read

def main():
    nombre = read(int)
    while nombre is not None:
        copia_nombre = nombre
        digits = 0
        while copia_nombre != 0:
            digits += copia_nombre % 10
            copia_nombre = copia_nombre // 10
        print("La suma dels digits de ", nombre, " es ", digits, ".", sep='')
        nombre = read(int)
main()
